plugins/workload-management/src/main/java/org/opensearch/plugin/wlm/WorkloadManagementPlugin.java

```diff
@@ -99,21 +106,29 @@ public Collection<Object> createComponents(
         IndexNameExpressionResolver indexNameExpressionResolver,
         Supplier<RepositoriesService> repositoriesServiceSupplier
     ) {
-        RulePersistenceServiceHolder.rulePersistenceService = new IndexStoredRulePersistenceService(
+        featureType = new WorkloadGroupFeatureType(new WorkloadGroupFeatureValueValidator(clusterService));
+        rulePersistenceService = new IndexStoredRulePersistenceService(
             INDEX_NAME,
             client,
+            clusterService,
             MAX_RULES_PER_PAGE,
-            new XContentRuleParser(WorkloadGroupFeatureType.INSTANCE),
+            new XContentRuleParser(featureType),
             new IndexBasedRuleQueryMapper()
         );
+        ruleRoutingService = new WorkloadGroupRuleRoutingService(client, clusterService);
         InMemoryRuleProcessingService ruleProcessingService = new InMemoryRuleProcessingService(
-            WorkloadGroupFeatureType.INSTANCE,
+            featureType,
             DefaultAttributeValueStore::new
         );
         autoTaggingActionFilter = new AutoTaggingActionFilter(ruleProcessingService, threadPool);
         return Collections.emptyList();
     }
 
+    @Override
+    public void getTransportService(TransportService transportService) {
+        ruleRoutingService.setTransportService(transportService);
```

Comment:
Better to do `((WorkloadGroupRuleRoutingService) ruleRoutingService).setTransportService` here to avoid having this specific method at the interface level.